test/prism/version_test.rb

```diff
@@ -4,8 +4,46 @@
 
 module Prism
   class VersionTest < TestCase
-    def test_version_is_set
+    def test_prism_version_is_set
       refute_nil VERSION
     end
+
+    def test_syntax_versions
+      assert Prism.parse("1 + 1", version: "3.3.0").success?
+      assert Prism.parse("1 + 1", version: "3.3.1").success?
+      assert Prism.parse("1 + 1", version: "3.3.9").success?
+      assert Prism.parse("1 + 1", version: "3.3.10").success?
+
+      assert Prism.parse("1 + 1", version: "3.4.0").success?
+      assert Prism.parse("1 + 1", version: "3.4.9").success?
+      assert Prism.parse("1 + 1", version: "3.4.10").success?
+
+      assert Prism.parse("1 + 1", version: "latest").success?
+
+      # Test edge case
+      error = assert_raise ArgumentError do
+        Prism.parse("1 + 1", version: "latest2")
+      end
+      assert_equal "invalid version: latest2", error.message
+
+      assert_raise ArgumentError do
+        Prism.parse("1 + 1", version: "3.3.a")
+      end
+
+      # Not supported version syntax
+      assert_raise ArgumentError do
+        Prism.parse("1 + 1", version: "3.3")
+      end
```

Comment:
Why not supported?
Won't 3.3 be expected as the latest of 3.3.x, or are you considering syntax changes across teeny versions?
IIRC, we have no such syntax changes (probably except for critical bugs, and in these years at least).

Reply:
I meant mostly "not supported at the moment/yet", only x.y.z for now.

> are you considering syntax changes across teeny versions?

Yes, at least the intention for now is to try to be careful about that.
So Prism matches exactly what the CRuby compiler accepts when using `Prism.parse(code, version: RUBY_VERSION)`, no more, no less.

IIRC there were a couple syntax changes in teeny versions a while ago.
OTOH https://github.com/whitequark/parser/tree/master/lib/parser only has x.y versions, so probably those are small changes.
I am not sure whether such changes can matter or not in practice, I suppose it depends on the tool.
For IRB for example it seems best to be exactly the same including "bugs" so IRB and `ruby -e` are equivalent.

I'd be OK to treat 3.3 as 3.3.LAST or so, but it feels better to do separately and it needs @kddnewton's opinion on this.